File: 4-Polynomial-Regression-Python-and-R-Language/polynomial_regression_sami.R
                         # Polynomial regression
# Shift + Ctrc + C is used to comment multiple lines 
# Importing Dataset Position_Salaries.csv 
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[2:3]

# Fitting Linear Regression to the dataset
lin_reg = lm(formula = Salary ~ . ,
             data = dataset)

# Fitting Polynomial Regression to the dataset
dataset$Level2 = dataset$Level^2
dataset$Level3 = dataset$Level^3
dataset$Level4 = dataset$Level^4
poly_reg = lm(formula = Salary ~ . ,
              data = dataset)

# Visualizing the Linear Regression results
library(ggplot2)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), color = 'red') +
  geom_line(aes(x = dataset$Level, y = predict(lin_reg, newdata = dataset)), color = 'blue ') +
  ggtitle('Truth or Bluff (Linear Regression)') +
  xlab('Level') +
  ylab('Salary')

# Visualizing the Polynomial Regression results
library(ggplot2)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary), color = 'red') +
  geom_line(aes(x = dataset$Level, y = predict(poly_reg, newdata = dataset)), color = 'blue ') +
  ggtitle('Truth or Bluff (Polynomial Regression)') +
  xlab('Level') +
  ylab('Salary')

# Predicting a new result with Linear Regression i.e level 6.5
y_pred_lin = predict(lin_reg, data.frame(Level = 6.5))

# Predicting a new result with Polynomial Regression i.e level 6.5
y_pred_poly = predict(poly_reg, data.frame(Level = 6.5,
                                      Level2 = 6.5^2,
                                      Level3 = 6.5^3,
                                      Level4 = 6.5^4))

File: 4-Polynomial-Regression-Python-and-R-Language/polynomial_regression_sami.py
                          # Polynomial Regression
# Import libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Importing dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2]
y = dataset.iloc[:, 2]

# Fitting Linear Regression to the dataset
from sklearn.linear_model import LinearRegression
lin_reg = LinearRegression()
lin_reg.fit(X, y)

#Fitting polynomial Regression to the dataset
from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 4)
X_poly = poly_reg.fit_transform(X)
lin_reg2 = LinearRegression()
lin_reg2.fit(X_poly, y)

# Visualising the Linear REgression results
plt.scatter(X, y, color = 'red')
#plt.plot(X, lin_reg.predict(X), color = 'blue')
plt.plot(X['Level'], lin_reg.predict(X), color = 'blue')
plt.title('Truth or Bluff (Linear regression result)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()

# Visualising the Polynomial Regression results
plt.scatter(X, y, color = 'red')
plt.plot(X['Level'], lin_reg2.predict(poly_reg.fit_transform(X)), color = 'blue')
plt.title('Truth or Bluff (Polynomial Regression)')
plt.xlabel('Position Level')
plt.ylabel('Salary')
plt.show()

# Predicting a new result with Linear Regression
print("level 6.5 prediction of linear regression is {0}", lin_reg.predict([[6.5]]))

# Predicting a new result with Polynomial Regression
#print("level 6.5 prediction of linear regression is {0}", lin_reg2.predict(poly_reg.fit_transform(6.5)))
print("level 6.5 prediction of linear regression is {0}", lin_reg2.predict(poly_reg.fit_transform([[6.5]])))















